erc20/contract/proxy.sol

```diff
@@ -0,0 +1,35 @@
+pragma solidity ^0.5.0;
+
+import "./token_storage.sol";
+
+// inherit from TokenStorage so we have the constructor, since the token variables need to be stored in the
+// proxy's storage
+contract Proxy is TokenStorage {
+    function () external payable {
+        // directly get the implementation contract address from the storage. This way we don't need to depend
+        // on the upgradeable contract
+        address _impl = getAddress(keccak256(abi.encode("implementation")));
+        require(_impl != address(0), "The implementation address can't be the zero address");
+        bytes memory data = msg.data;
+
+        assembly {
+            let result := delegatecall(gas, _impl, add(data, 0x20), mload(data), 0, 0)
+            let size := returndatasize
+            let ptr := mload(0x40)
+            returndatacopy(ptr, 0, size)
+            switch result
+            case 0 { revert(ptr, size) }
+            default { return(ptr, size) }
+        }
+    }
+
+    constructor() public {
+        //set initial contract address, needs to be hardcoded
+        // TODO: Set correct address
```

Comment:
Is there an issue about this open @LeeSmet? I would like to make a habit of linking TODO's to issues.

Reply:
No issue open, however this value will be different for both mainnet and testnet, and for devnet as well. So it's probably better to just leave this empty I guess? and maybe add a `require(addr != address(0))` to cause deployment to fail at runtime if this is forgotten

Reply:
The require I would definitely do. And please do create an issue for defining the addresses at least, and link it to the ERC20 bridge story.

Reply:
Issue created: #260